### PR TITLE
[update] support armclang

### PR DESCRIPTION
**Pull request by liukangcc**
## 拉取/合并请求描述：(PR description)

[
这段方括号里的内容是您**必须填写并替换掉**的，否则PR不可能被合并。**方括号外面的内容不需要修改，但请仔细阅读。**
The content in this square bracket must be filled in and replaced, otherwise, PR can not be merged. The contents outside square brackets need not be changed, but please read them carefully.

请在这里填写您的PR描述，可以包括以下之一的内容：为什么提交这份PR；解决的问题是什么，你的解决方案是什么；
Please fill in your PR description here, which can include one of the following items: why to submit this PR; what is the problem solved and what is your solution;

并确认并列出已经在什么情况或板卡上进行了测试。
And confirm in which case or board has been tested.
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in the submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use a web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality
- [x] 本拉取/合并使用[formatting](https://github.com/mysterywolf/formatting)等源码格式化工具确保格式符合[RT-Thread代码规范](../documentation/coding_style_cn.md) This PR complies with [RT-Thread code specification](../documentation/coding_style_en.txt)

### CODE REVIEW

**Review by Guozhanxin — APPROVED** [review]

File: tools/building.py [hunk 1]
```diff
@@ -840,6 +841,10 @@ def GenTargetProject(program = None):
         from keil import MDK5Project
         MDK5Project('project.uvprojx', Projects)
 
+    if GetOption('target') == 'mdk5-ac6':
```
Comment (Guozhanxin) [inline tools/building.py:844]: 这个名字改成 mdk-ac6 吧

**Review by Guozhanxin — CHANGES_REQUESTED** [review]
最后记得把模板也更新了 https://github.com/RT-Thread/rt-thread/tree/master/bsp/stm32/libraries/templates

**Comment by mysterywolf** [comment]
waitting for https://github.com/RT-Thread/rt-thread/pull/5008 and https://github.com/RT-Thread/rt-thread/pull/5003

**Review by mysterywolf — APPROVED** [review]

File: bsp/stm32/stm32l4r9-st-sensortile-box/rtconfig.py [hunk 2]
```diff
@@ -28,6 +28,9 @@
 if os.getenv('RTT_EXEC_PATH'):
     EXEC_PATH = os.getenv('RTT_EXEC_PATH')
 
+if os.getenv('RTT_PLATFORM'):
+    PLATFORM = os.getenv('RTT_PLATFORM')
```
Comment (BernardXiong) [inline bsp/stm32/stm32l4r9-st-sensortile-box/rtconfig.py:32]: Why use `RTT_PLATFORM` env value? There is a env value named as `RTT_CC`, which is related with platform.

Comment (Guozhanxin) [reply bsp/stm32/stm32l4r9-st-sensortile-box/rtconfig.py:32]: 之前的用法，设定的 RTT_CC= keil ，但是，这样仍然不知道是使用 armcc 还是 armclang，所以，以后应该设定 RTT_CC=armcc 吗？

Comment (BernardXiong) [reply bsp/stm32/stm32l4r9-st-sensortile-box/rtconfig.py:32]: 那需要对RTT_CC做个调整，而不是armcc了，而是RTT_CC=armclang

Comment (liukangcc) [reply bsp/stm32/stm32l4r9-st-sensortile-box/rtconfig.py]: 目前的修改，不需要设置 `RTT_PLATFORM` ， 当用户想使用 `armclang ` 时，需要手动修改 `PLATFORM    = 'armclang'`

File: tools/building.py [hunk 3]
```diff
@@ -259,6 +259,7 @@ def PrepareBuilding(env, root_directory, has_libcpu=False, remove_components = [
                 'ses' : ('gcc', 'gcc'),
                 'cmake':('gcc', 'gcc'),
                 'cmake-armclang':('keil', 'armclang'),
+                'mdk-ac6': ('keil', 'armclang'),
```
Comment (Guozhanxin) [inline tools/building.py:262]: 既然这样还是去掉这个命令吧，如果他们想生产ac6的工程，就让他们自己改模板

**Review by Guozhanxin — CHANGES_REQUESTED** [review]
既然这样还是去掉这个命令吧，如果他们想生产ac6的工程，就让他们自己改模板

**Review by Guozhanxin — APPROVED** [review]

**Comment by Guozhanxin** [comment]
commit 个数太多了，合并一下吧

**Pull request merged**